userspace/falco/configuration.h

```diff
@@ -277,48 +281,60 @@ class falco_configuration
 
 namespace YAML {
 	template<>
-	struct convert<falco_configuration::plugin_config> {
-
-		static bool read_file_from_key(const Node &node, const std::string &prefix, std::string &value)
+	struct convert<nlohmann::json> {
+		static bool decode(const Node& node, nlohmann::json& res)
 		{
-			std::string key = prefix;
-
-			if(node[key])
-			{
-				value = node[key].as<std::string>();
-				return true;
-			}
-
-			key += "_file";
-
-			if(node[key])
-			{
-				std::string path = node[key].as<std::string>();
-
-				// prepend share dir if path is not absolute
-				if(path.at(0) != '/')
-				{
-					path = string(FALCO_ENGINE_PLUGINS_DIR) + path;
-				}
-
-				// Intentionally letting potential
-				// exception be thrown, will get
-				// caught when reading config.
-				std::ifstream f(path);
-				std::string str((std::istreambuf_iterator<char>(f)),
-						std::istreambuf_iterator<char>());
-
-				value = str;
-				return true;
+			int int_val;
+			double double_val;
+			bool bool_val;
+			std::string str_val;
+			nlohmann::json sub{};
+
+			switch (node.Type()) {
+				case YAML::NodeType::Map:
+					for (auto &&it: node)
+					{
+						YAML::convert<nlohmann::json>::decode(it.second, sub);
```

Comment:
Not that any legitimate use would do this, but we might want to cap the recursion at some big but fixed number of levels.